tests/python/pants_test/task/test_task.py

```diff
@@ -95,15 +105,17 @@ def test_incremental(self):
     vtB = task.execute()
     self.assertContent(vtB, one + two)
 
-    # Incremental atop existing directory for vtA.
+    # vtC.previous_cache_key == vtB.cache_key so the task appends the file to vtB's results.
+    # This behavior used to be skipped because the current_dir existed when checked by cache_manager._use_previous_dir.
```

Comment:
I am tinkering with this now...I think I can break out the previous_results move to happen after the cache check. It is tough to reason about the cache manager code, I have written another followup that deletes a lot (but not all) of the circular referencing and object mutation.

The above test should currently pass (although maybe it doesn't? I am working on a different integration test failure atm). The case it presents is no longer very useful though. Can you elaborate on what you are asking for with: 

"I would still love to see the test updated before landing this though."